src/backend/joanie/core/utils/payment_schedule.py

```diff
@@ -150,3 +152,28 @@ def has_installments_to_debit(order):
     return any(
         is_installment_to_debit(installment) for installment in order.payment_schedule
     )
+
+
+def convert_due_date_to_date_object(item: dict):
+    """
+    Converts the `due_date` string into a date object.
+    """
+    try:
+        item["due_date"] = date.fromisoformat(item["due_date"])
+    except ValueError as exception:
+        raise InvalidConversionError(
+            f"Invalid date format for due_date: {exception}."
+        ) from exception
```

Comment:
I prefer pure functions:
```suggestion
def convert_date_str_to_date_object(date_str: str):
    """
    Converts a `date_str` string into a date object.
    """
    try:
        return date.fromisoformat(due_date)
    except ValueError as exception:
        raise InvalidConversionError(
            f"Invalid date format for date_str: {exception}."
        ) from exception
```